client/src/setup/setupLanguageServerHooks.ts

```diff
@@ -92,12 +93,16 @@ const startLanguageServer = (extensionState: ExtensionState): void => {
     notifyTelemetryChanged
   );
 
-  const hardhatTelemetryChangeDisposable = workspace.onDidChangeConfiguration(
-    notifyTelemetryChanged
-  );
+  const configChangedDisposable = workspace.onDidChangeConfiguration(() => {
```

Comment:
The idea here is to push the entire config to the language server? Rather than tracking individual config changes?

Reply:
Yes, the entire config is synced to the server

Reply:
Can I confirm I have understood the code. The complete extension config is synced, but we retain the previous approach to telemetry enabled (which I think is fine - I just want to double check my mental model)?

Reply:
In an out of band discussion, the telemetry update at the hardhat level has been subsumed within extension config updates, but the global setting still remains separate.